api/src/main/java/ca/bc/gov/educ/api/gradstudent/service/GradBaseService.java

```diff
@@ -24,4 +40,60 @@ protected void validateStudentStatusAndResetBatchFlags(GraduationStudentRecordEn
             gradEntity.setRecalculateProjectedGrad(null);
         }
     }
+
+    protected List<School> getSchoolsByDistricts(List<UUID> districtIds, String accessToken) {
+        List<School> results = new ArrayList<>();
+        for (UUID districtId : districtIds) {
+            results.addAll(getSchoolsByDistrictId(districtId, accessToken));
+        }
+        return results;
+    }
+
+    protected List<School> getSchoolsByDistrictId(UUID districtId, String accessToken) {
+        return getWebClient().get()
+                .uri(String.format(getConstants().getSchoolsByDistrictIdUrl(), districtId))
+                .headers(h -> {
+                    h.setBearerAuth(accessToken);
```

Comment:
we shouldn't be using access token anymore

Reply:
This can be handled in a new ticket.